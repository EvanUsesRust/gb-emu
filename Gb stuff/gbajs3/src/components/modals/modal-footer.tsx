import { FooterWrapper } from '../shared/styled.tsx';

import type { ReactNode } from 'react';

type ModalFooterProps = {
  children: ReactNode;
};

export const ModalFooter = ({ children }: ModalFooterProps) => {
  return (
    <FooterWrapper data-testid="modal-footer:wrapper">{children}</FooterWrapper>
  );
};
