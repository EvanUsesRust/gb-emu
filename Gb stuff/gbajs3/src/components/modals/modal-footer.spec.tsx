import { screen } from '@testing-library/react';
import { describe, expect, it } from 'vitest';

import { ModalFooter } from './modal-footer.tsx';
import { renderWithContext } from '../../../test/render-with-context.tsx';

describe('<ModalFooter />', () => {
  it('renders children', () => {
    renderWithContext(<ModalFooter>Test Footer Content</ModalFooter>);

    expect(screen.getByText('Test Footer Content')).toBeInTheDocument();
  });

  it('container matches snapshot', () => {
    renderWithContext(<ModalFooter>Test Footer Content</ModalFooter>);

    expect(screen.getByTestId(`modal-footer:wrapper`)).toMatchSnapshot();
  });
});
