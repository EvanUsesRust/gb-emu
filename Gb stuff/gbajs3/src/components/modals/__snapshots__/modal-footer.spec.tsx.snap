// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<ModalFooter /> > container matches snapshot 1`] = `
.c0 {
  display: flex;
  flex-direction: row;
  gap: 10px;
  align-items: center;
  justify-content: flex-end;
  border-top: 1px solid #dee2e6;
  padding: 1rem 1rem;
}

<div
  class="c0"
  data-testid="modal-footer:wrapper"
>
  Test Footer Content
</div>
`;
